x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/analysis/analyzer/Analyzer.java

```diff
@@ -1312,4 +1303,17 @@ protected boolean skipResolved() {
             return true;
         }
     }
+    
+    abstract static class LogicalPlanAnalyzeRule extends AnalyzeRule<LogicalPlan> {
+
+        @Override
+        protected LogicalPlan rule(LogicalPlan plan) {
+            if (plan.childrenResolved() == false) {
+                return plan;
+            }
+            return rulePlan(plan);
+        }
+
+        protected abstract LogicalPlan rulePlan(LogicalPlan plan);
```

Comment:
`rulePlan` is repetitive (the method only accepts `LogicalPlan`s)  - how about `doRule` or `analyze`?